### PR TITLE
[speechx] add batch recognizer decode.

### PR DESCRIPTION
**Pull request by SmileGoat**
### PR types

New features
### PR changes

Others
### Describe

add batch recognizer decode, multi-thread recognizer decode.

### CODE REVIEW

File: speechx/speechx/asr/recognizer/u2_recognizer.h [hunk 1]
```diff
@@ -111,7 +111,9 @@ struct U2RecognizerResource {
 
 class U2Recognizer {
   public:
-    explicit U2Recognizer(const U2RecognizerResource& resouce);
+    U2Recognizer(const U2RecognizerResource& resouce);
```
Comment (zh794390558) [inline speechx/speechx/asr/recognizer/u2_recognizer.h]: 加上explicit

File: speechx/speechx/asr/recognizer/u2_recognizer.cc [hunk 2]
```diff
@@ -46,6 +46,29 @@ U2Recognizer::U2Recognizer(const U2RecognizerResource& resource)
 
 }
 
+U2Recognizer::U2Recognizer(const U2RecognizerResource& resource, 
+                           std::shared_ptr<NnetBase>nnet)
+    : opts_(resource) {
+    BaseFloat am_scale = resource.acoustic_scale;
+    const FeaturePipelineOptions& feature_opts = resource.feature_pipeline_opts;
+    std::shared_ptr<FeaturePipeline> feature_pipeline(
```
Comment (zh794390558) [inline speechx/speechx/asr/recognizer/u2_recognizer.cc]: 用make_shared

File: speechx/speechx/asr/recognizer/u2_recognizer.cc [hunk 3]
```diff
@@ -46,6 +46,29 @@ U2Recognizer::U2Recognizer(const U2RecognizerResource& resource)
 
 }
 
+U2Recognizer::U2Recognizer(const U2RecognizerResource& resource,
```
Comment (zh794390558) [inline speechx/speechx/asr/recognizer/u2_recognizer.cc]: U2RecognizerResource 更像是Options

File: speechx/speechx/asr/recognizer/u2_recognizer_batch_main.cc [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                     std::vector<string> uttlist,
+                     std::vector<string>* results) {
+    int32 num_done = 0, num_err = 0;
+    double tot_wav_duration = 0.0;
+    double tot_attention_rescore_time = 0.0;
+    double tot_decode_time = 0.0;
+    int chunk_sample_size = FLAGS_streaming_chunk * FLAGS_sample_rate;
+    if (wavlist.empty()) return;
+
+    std::shared_ptr<ppspeech::U2Recognizer> recognizer_ptr(
```
Comment (zh794390558) [inline speechx/speechx/asr/recognizer/u2_recognizer_batch_main.cc]: 多用make_shared

File: speechx/speechx/asr/recognizer/u2_recognizer_batch_main.cc [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    LOG(INFO) << "chunk size (sample): " << chunk_sample_size;
+
+    ppspeech::U2RecognizerResource resource =
+        ppspeech::U2RecognizerResource::InitFromFlags();
+    ThreadPool threadpool(njob);
+    vector<vector<string>> wavlist;
+    vector<vector<string>> uttlist;
+    vector<vector<string>> resultlist(njob);
+    vector<std::future<void>> futurelist;
+    std::thread threads[njob];
```
Comment (zh794390558) [inline speechx/speechx/asr/recognizer/u2_recognizer_batch_main.cc]: del

**Review by zh794390558 — COMMENTED** [review]
过下pre-commit

File: speechx/speechx/asr/recognizer/u2_recognizer.cc (hunk 3 above)
Comment (SmileGoat) [reply speechx/speechx/asr/recognizer/u2_recognizer.cc]: 这个后面会改；现在先不改；后面会统一refactor

**Pull request merged**